[John Chia Heng Kit] iP

Johnchiahk opened the pull request:
# Duke project template

> "Your mind is for having ideas, not holding them." - David Allen [(source)](https://dansilvestre.com/productivity-quotes/)

Duke frees your mind of having to remember things you need to do. It's 

- text-based
- easy to learn
- ~~FAST~~ *SUPER* FAST to use

All you need to do is, 

1. download it from [here](https://nus-cs2103-ay2223s2.github.io/website/schedule/week4/project.html)
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 👍

And it is **FREE**!

Features:
- [x] Managing tasks
- [X] Managing deadlines
- [ ] Reminders (coming soon)
---
If you Java programmer, you can use it to practice Java too. Here's the main method:
```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

Johnchiahk committed:
Duke.java added from ip folder

Johnchiahk committed:
"Level-2 by John Chia"

Johnchiahk committed:
"Task class implemented!"

Johnchiahk committed:
Deadline, Event, Todo subclasses of Task Class added!

Johnchiahk committed:
EXPECTED.TXT, input.txt and runtest.bat.txt added to faciliate automated Text UI Testing

Johnchiahk committed:
DukeException class created.

Johnchiahk committed:
try catch blocks created for Todo, Deadline, and Event classes to handle DukeExceptions.

Johnchiahk committed:
deleteTask function in Duke Class completed, with exceptions handled for the "delete" case.

Johnchiahk committed:
Save the tasks in the hard disk automatically whenever the task list changes.

Load the data from the hard disk when Duke starts up.
A class, TaskManagement, is created to store the list
of tasks in the file tasks.txt.

Johnchiahk committed:
TaskStorage class created to encapsulate a taskStorage object.

Johnchiahk committed:
TaskStorage class, which encapsulates a taskStorage object, is created in master branch.

Johnchiahk committed:
Add LocalDate class in both Deadline and Event class

Johnchiahk committed:
Merge branch 'branch-Level-7'

Johnchiahk committed:
Merge branch 'branch-Level-8'

Johnchiahk committed:
Merge remote-tracking branch 'origin/add-gradle-support'

Johnchiahk committed:
Make the code more OOP

Ui class added to encapsulate a User Interface.

Johnchiahk committed:
Create packages duke and duke.task

Johnchiahk committed:
Add DeadlineTest and EventTest

Gradle is used for efficient build automation

wengkit1 reviewed (commented):
Could exceptions be possibly extracted and handled in another package?  
Otherwise nice design lgtm!

sembcorpp commented:
Code is easy to read and follow but maybe some java docs would be good so its more clear what the method is doing and what params it takes in? But good job so far

kimberlybp commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,192 @@
+            inp = br.readLine();
+            String[] input = inp.split(" ");
+            ui.showLine();
+            switch(input[0]) {
+                case "list":
+                    System.out.println("Tasks:");
+                    //System.out.println(taskstorage);
+                    for (int i = 0; i < taskStorage.noTasks(); i++) {
+                        System.out.print(i + 1 + ".");
+                        System.out.println(taskStorage.getTask(i));
+                    }
+                    break;
+
+                case "bye":
+                    System.out.println("Byeee! Hope to see you again! Signing off, duke.Duke.");
+                    break;
+
```
Do remember that coding standards state that here should be no indentation for `case` clauses in switch statements

Johnchiahk replied:
Thanks I've rectified it

kimberlybp commented on src/main/java/duke/TaskStorage.java:
```diff
@@ -0,0 +1,38 @@
+        taskstorage = new ArrayList<Task>();
+    }
+    public void addTask(Task t) {
+        System.out.println("Got it. I've added this task:");
+        System.out.println("  " + t);
+        this.taskstorage.add(t);
+        ind++;
+        System.out.println("Now you have " + ind + " task(s) in the list.");
```
Perhaps you can move System.out.println(..) into a function in your Ui class to utilise it more, since the purpose of the Ui class is to handle all interactions with the user

Johnchiahk replied:
Noted with thanks!

kimberlybp commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,192 @@
+        StringBuilder sb = new StringBuilder();
+
+        ui.showWelcome();
+//        String logo = " ____        _        \n"
+//                + "|  _ \\ _   _| | _____ \n"
+//                + "| | | | | | | |/ / _ \\\n"
+//                + "| |_| | |_| |   <  __/\n"
+//                + "|____/ \\__,_|_|\\_\\___|\n";
+        String line = "-------------------------------";
+//        System.out.println(line);
+//        System.out.println("Hiii Im\n" + logo);
+//        System.out.println("What can I do for you hmm?");
+//        System.out.println(line);
```
A small suggestion: would recommend removing commented, unused lines of code (for cleaner code)

Johnchiahk replied:
Thank you!

kimberlybp reviewed (commented):
Overall, code looks good with some small issues

wengkit1 commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,192 @@
+                        break;
+                    }
+
+                default:
```
good to include default behaviour, consistent with coding standards.

Johnchiahk replied:
Alright thank you!

wengkit1 commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,192 @@
+                        System.out.println("Number should be typed in");
+                    } catch (DukeException e){
+                        System.out.println(e.getMessage());
+                    } finally {
```
Is a finally block necessary in this case?

Johnchiahk replied:
Hi wengkit! On second thought, the finally block is not necessary, but I believe the break statement clearly shows to the reader that each case will only be executed at most once 😅

Johnchiahk committed:
Update Parser class, more OOP Principles implemented

Johnchiahk committed:
Create JavaDocs for all methods in all classes.

Johnchiahk committed:
Create JavaDocs for Duke, Ui, and TaskStorage classes

Johnchiahk committed:
Merge branch 'A-JavaDoc'

Johnchiahk committed:
Apply switch case coding standard

Johnchiahk committed:
Merge branch 'A-CodingStandard'

Johnchiahk committed:
Update README.md

Johnchiahk committed:
Update README.md

Johnchiahk committed:
Update README.md

Johnchiahk committed:
Implements GUI

Johnchiahk committed:
Merge branch 'master' of https://github.com/Johnchiahk/ip

Johnchiahk committed:
Remove hunks of commented out code for all files

Johnchiahk committed:
Update README.md

Johnchiahk committed:
Adds DukeException case to find function

DukeException handles all exceptions specifically to Duke.

Johnchiahk committed:
Merge remote-tracking branch 'origin/master'

Johnchiahk committed:
Merge tag 'Level-9'

Johnchiahk committed:
Implement Javadocs correctly

Johnchiahk committed:
Implements SLAP on the execute function

The execute function code is too long.

Needs to be shortened.

Implements SLAP to make it programmer-friendly.

Johnchiahk committed:
Put binary files in .gitignore

Johnchiahk committed:
change .gitignore back to original

Johnchiahk committed:
Omit binary files in repo

Johnchiahk committed:
Update README.md

Johnchiahk committed:
Update README.md

Johnchiahk committed:
Update README.md

Johnchiahk committed:
Update README.md

Johnchiahk committed:
Update README.md

Johnchiahk committed:
Update README.md

Johnchiahk committed:
Update README.md

Johnchiahk committed:
Update README.md

Johnchiahk committed:
Update README.md

Johnchiahk committed:
Update README.md

Use GFMD

Johnchiahk committed:
Update README.md

Johnchiahk committed:
Update README.md

Johnchiahk committed:
Update README.md

Johnchiahk committed:
Update README.md

Johnchiahk committed:
Update README.md

Johnchiahk committed:
Update README.md

Johnchiahk committed:
Add assert to deadline and event function in Parser class

Johnchiahk committed:
Merge branch 'master' of https://github.com/Johnchiahk/ip

Johnchiahk committed:
Merge pull request #2 from Johnchiahk/branch-A-Assertions

Add assert to deadline and event function in Parser class

Johnchiahk committed:
Merge branch 'master' of https://github.com/Johnchiahk/ip

Johnchiahk committed:
Create SortTaskByName and sort method in Parser

Johnchiahk committed:
Delete A class, add javadocs to SortTaskByName

Johnchiahk committed:
Update J-Unit Tests

Johnchiahk committed:
Enhance GUI

Johnchiahk committed:
Upload UI picture

Johnchiahk committed:
Update Ui.png

Johnchiahk committed:
Update README.md

Johnchiahk committed:
Update README.md

Johnchiahk committed:
Update README.md

Johnchiahk committed:
Update README.md

Johnchiahk committed:
Update README.md

Johnchiahk committed:
Update README.md

Johnchiahk committed:
Update README.md

Johnchiahk committed:
Update README.md

Johnchiahk committed:
Update README.md

Johnchiahk committed:
Update README.md

Johnchiahk committed:
Update README.md

Johnchiahk committed:
Update README.md

Johnchiahk committed:
Update README.md

Johnchiahk committed:
Update README.md

Johnchiahk committed:
Update README.md

Johnchiahk committed:
Update README.md

Johnchiahk committed:
Update README.md

Johnchiahk committed:
Update README.md

Johnchiahk committed:
Update README.md

Johnchiahk committed:
Update README.md

Johnchiahk committed:
Update README.md

Johnchiahk committed:
Update README.md

Johnchiahk committed:
Update README.md

Johnchiahk committed:
Merge branch 'master' of https://github.com/Johnchiahk/ip

Johnchiahk committed:
Update final version

Johnchiahk committed:
Create retrieve function but not done

Johnchiahk committed:
Complete retrieve function

When duke application is relaunched, the data from the tasklist in the previous application  will be
loaded onto the application

Johnchiahk committed:
Complete load function

Johnchiahk committed:
Apply SLAP on TaskManagement

Johnchiahk committed:
Update JUnit tests, remove binary files in repo

Johnchiahk committed:
Remove binary files from repo

Johnchiahk committed:
Remove src/out/ binary files in repo

Johnchiahk committed:
Delete binary files in repo v1.3
